utils/S3UtilsMongoClient.js

```diff
@@ -111,6 +197,14 @@ class S3UtilsMongoClient extends MongoClientInterface {
                 return callback(errors.InternalError);
             }
 
+            const bucketEntry = Object.keys(usersBucketCreationDatesMap).find(id => id.endsWith(`${constants.splitter}${bucketName}`));
+            if (bucketEntry) {
+                bucketKey = `bucket_${bucketName}_${new Date(usersBucketCreationDatesMap[bucketEntry]).getTime()}`;
+                if (bucketKey) {
```

Comment:
the keys in `userBucketCreationDatesMap` are of the form `${ownerId}${constants.splitter}${bucketName}` : we now bucketName, is the ownerId not part of `bucketInfo` as well, as `bucketInfo.ownerCanonicalId` ?

this would allow the inefficient search into the list...

Reply:
do you mean reusing this info directly to reduce I/Os?

Reply:
i mean this should work (to avoid the lineaer search):


```suggestion
            const bucketEntry = usersBucketCreationDatesMap[`${bucketInfo.ownerCanonicalId}${constants.splitter}${bucketName}`];
```

Reply:
Yes you are right, we can call the .getOwner() to get the key directly => code updated